src/sentry/middleware/proxy.py

```diff
@@ -72,6 +77,31 @@ def process_request(self, request):
             request.META['CONTENT_LENGTH'] = '4294967295'  # 0xffffffff
 
 
+COMPRESSED_ENCODINGS = ('gzip', 'deflate')
+
+
+def decompress(data, encoding):
+    if encoding == 'gzip':
+        return zlib.decompress(data, zlib.MAX_WBITS | 16)
+    if encoding == 'deflate':
+        return zlib.decompress(data, -zlib.MAX_WBITS)
+
+
+class DecompressBodyMiddleware(object):
+    def process_request(self, request):
+        encoding = request.META.get('HTTP_CONTENT_ENCODING', '').lower()
+        if encoding in COMPRESSED_ENCODINGS:
+            data = request._stream.read()
+            try:
+                uncompressed = decompress(data, encoding)
+                request._stream = six.StringIO(uncompressed)
+                request.META['CONTENT_LENGTH'] = len(uncompressed)
+                request.META['HTTP_CONTENT_ENCODING'] = 'identity'
```

Comment:
Wondering if it's better to just remove this here, instead.

Reply:
If we do this at this level I would opt for patching our django to support `wsgi.input_terminated` and have the input stream decode on the fly (there is a gzip file that can do that). We already support reading to the terminator for the chunked request case so it should not be much of a change.

Reply:
That for the record means setting the content length to empty string.